### PR TITLE
multiple tag support to allow use in non-trivial XML files

### PR DESCRIPTION
**Pull request by markdicksonjr**

### CODE REVIEW

**Comment by pirxpilot** [comment]
This change is not backward compatible (which might be OK if we release a new major version).  I wonder if there is a way to implement it to preserve it though. Option part seems to be trivial: `tag` could accept both a single tag name (String) and an Array of tag names.
Not sure about the format of the emitted object though. We could emit { record, tagName } if there were multiple tags used and { ...record } if only a single tag was used, but that seems a bit too clever.

File: lib/xmlnode.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ XmlNode.prototype.createSaxParser = function (options) {
     if (record) {
       record = elem.addChild(record, node.name);
     }
-    else if (options.tags.indexOf(node.name) !== -1 && !currentTag) {
+    else if (options.tag === node.name || Array.isArray(options.tag) && options.tag.indexOf(node.name) !== -1 && !currentTag) {
```
Comment (pirxpilot) [inline lib/xmlnode.js]: This one is tricky: `&&` [binds stronger][1] than `||` - in other words `a || b && c` is not the same as `(a || b) && c` which we need here. Strange that unit tests didn't fail on that.
More importantly though the `isArray` check should probably be done only once. One solution is to define 2 functions to check for matching tag name and switch between implementation for an Array and for a single tag as soon as we determine if we are dealing with an Array or not.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence

**Comment by pirxpilot** [comment]
Please check out my changes in [pr4] branch. Looks lie match functions can be a bit simpler (I left the tests you added and they all seem to pass). I also kept the original data format when only a single tag is specified. If I missed something let me know.
If those work for you, I'll release a new version of the module.
Feel free to squash the history. 

[pr4]: https://github.com/melitele/sax-stream/commits/pr4

**Comment by markdicksonjr** [comment]
That's definitely simpler :)

**Comment by pirxpilot** [comment]
Thanks. Released as part of 1.3.0

**Pull request closed without merging**